community/kernel/src/main/java/org/neo4j/kernel/impl/api/DefaultTransactionTracer.java

```diff
@@ -177,4 +194,22 @@ public long logRotationAccumulatedTotalTimeMillis()
     {
         return TimeUnit.NANOSECONDS.toMillis( accumulatedTotalTimeNanos.get() );
     }
+
+    @Override
+    public void register( Observer observer )
```

Comment:
i have quite a lot of concerns about why we need to introduce observers here, lets take if offline. Will you have couple of minutes to talk?